### PR TITLE
Feature 2 : Profiles Listing Page with dynamic data

### PR DESCRIPTION
**Pull request by Abhishek-Jain-1925**
Feature 2 : Profiles Listing Page with dynamic data

Work done are as follows - 
1.Complete Listing Page in Tabular Format
2.Dynamic data fetching from GO backend with the help of AXOIS
3.Data Formating and Validation

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { useState } from "react";
+import { ResumeContext } from "./utils/ResumeContext";
+import "./App.css";
+import Router from "./routes/router";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+
+const App = () => {
+  const queryClient = new QueryClient();
+  const [initialState, setInitialState] = useState({
```
Comment (Payal13) [inline src/App.jsx:9]: can we add redux to maintain data ? as we need to use data fetched from API at multiple places.

https://redux.js.org/introduction/getting-started

File: src/Constants.js [hunk 2]
```diff
@@ -28,6 +46,11 @@ export const skillsOptions = [
   { label: "VueJs", value: "VueJs" },
 ];
 
+export const ROUTES = {
```
Comment (Payal13) [inline src/Constants.js:49]: please add a separete files for routes path constants
e.g - 
export const PROFILE_API_URL = '/api/profile/';

File: src/components/Builder/BasicInfo.jsx
Comment (Payal13) [inline src/components/Builder/BasicInfo.jsx:1]: why jsx ?

File: src/Constants.js [hunk 3]
```diff
@@ -1,11 +1,29 @@
...
+
+export const GENDER = [
   { label: "MALE", value: "Male" },
   { label: "FEMALE", value: "Female" },
   { label: "OTHER", value: "Other" },
 ];
-//ReactJs, Angular, Flutter, .Net, NodeJs, React-Native, VueJs
-export const skillsOptions = [
+
+export const DESIGNATION = [
```
Comment (Payal13) [inline src/Constants.js:15]: Please check this again, i think value and label needs to be interchanged.
and so many spelling mistakes are there.

File: src/components/Builder/Builder.module.css
Comment (Payal13) [inline src/components/Builder/Builder.module.css:1]: Try not to add explicite css file. use and components properly

File: src/components/Builder/Index.jsx
Comment (Payal13) [inline src/components/Builder/Index.jsx:1]: keep uniformity in file names. Index.jsx or index.js
mostly use index.js

each of this should have be folder and and index file - 
e.g .../Certification/index.js
so you can add other helper files for Certification in same folder.

update all as above

File: src/services/axios.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+export const post = async (url, payload) => {
+  try {
+    const response = await client.post(url, payload);
+    return response.data;
+  } catch (error) {
+    throw new Error(error.message);
+  }
+};
+
+export const put = async (url) => {
```
Comment (Payal13) [inline src/services/axios.js:51]: put, patch, del etc will also have payload

File: src/utils/dto/constants.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
+import { Space, Tag } from "antd";
+
+export const columns = [
+  {
```
Comment (Payal13) [inline src/utils/dto/constants.js:4]: add Sr. No and remove database Id

File: src/utils/dto/constants.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    key: "years_of_experience",
+  },
+  {
+    title: "Primary Skills",
+    key: "primary_skills",
+    dataIndex: "primarySkills",
+    render: (_, { primary_skills }) => (
+      <>
+        {primary_skills.map((tag, index) => {
+          let color = tag.length > 5 ? "geekblue" : "green";
```
Comment (Payal13) [inline src/utils/dto/constants.js:33]: lets avoid this conditions. we can keep random color for now

File: src/components/Builder/Certification.jsx [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+  const onReset = () => {
+    form.resetFields();
+    setInitialState({
+      ...initialState,
+      certifications: {},
+    });
+  };
+
+  console.log("first");
```
Comment (Payal13) [inline src/components/Builder/Certification.jsx]: remove logs

File: src/App.jsx (hunk 1 above)
Comment (abhis-josh) [reply src/App.jsx:9]: Discussed Offline

Comment (Abhishek-Jain-1925) [reply src/App.jsx:9]: As per Discussed Offline

File: src/components/Builder/Certification.jsx (hunk 7 above)
Comment (Abhishek-Jain-1925) [reply src/components/Builder/Certification.jsx]: Removed All Logs from Project

File: src/services/axios.js (hunk 4 above)
Comment (Abhishek-Jain-1925) [reply src/services/axios.js:51]: As per discussed offline

File: src/components/Builder/Index.jsx
Comment (Abhishek-Jain-1925) [reply src/components/Builder/Index.jsx:1]: Updated Directory Structure for all files

**Pull request merged**